internal/locate/region_cache.go

```diff
@@ -121,6 +122,29 @@ func SetRegionCacheTTLSec(t int64) {
 	regionCacheTTLSec = t
 }
 
+// regionCacheTTLJitterSec is the max jitter time for region cache TTL.
+var regionCacheTTLJitterSec int64 = 60
+
+// SetRegionCacheTTLWithJitter sets region cache TTL with jitter. The real TTL is in range of [base, base+jitter).
+func SetRegionCacheTTLWithJitter(base int64, jitter int64) {
+	regionCacheTTLSec = base
+	regionCacheTTLJitterSec = jitter
+}
+
+// nextTTL returns a random TTL in range [ts+base, ts+base+jitter).
+func nextTTL(ts int64) int64 {
+	jitter := int64(0)
+	if regionCacheTTLJitterSec > 0 {
+		jitter = rand.Int63n(regionCacheTTLJitterSec)
+	}
+	return ts + regionCacheTTLSec + jitter
```

Comment:
Is there a way to add some constraint checks to make sure the unit of the input `ts` is `second`, to avoid being used mistakenly?

Reply:
We pass `ts` here to avoid overhead of `time.Now()`. To validate it, maybe we can:
- record process start ts by a global var (eg. processStartTs)
- the ts might be invalid if ts - processStartTs > 100 years or ts < processStartTs
- if the ts is invalid, we can fallback to call `time.Now()`

But do we really need to do that? There are many mechanisms to invalidate a stale region, it might be not a big problem even when we call `nextTTL` with epoch millis, I guess.

Or, we can just change the function signature (also many releated functions like `checkRegionCacheTTL`), let it (them) accept an arg of type `time.Time`.

Reply:
Maybe we could just add some comments to explain the constraints of the input `ts` in the comment if it's not a big problem when used mistakenly.